src/sa_web/static/js/routes.js

```diff
@@ -22,6 +22,27 @@ var Shareabouts = Shareabouts || {};
           startPageConfig,
           filteredRoutes;
 
+      this.collection = {
+        "landmark": {},
+        "place": {}
+      };
+      this.activities = {
+        "place": {}
+      };
```

Comment:
It's a bit subjective, but we may want to consider storing our landmarks and places in separate data structures. We can do this for our collections, activities, and configs. I think it would make the code more readable - but I might be missing something - so it's up to you. It would definitely break up the nested for-each loops, and it would make references like this: `self.collection[config.type][config.id] = collection;` easier to read, because the collections will be explicitly labelled and not nested.

I am thinking that we'll have `this.landmarkCollections`, `this.placeCollections`, `this.placeActivities`, `this.landmarkConfigs`, and `this.placeConfigs`. Of course, the downside is that we'll have more objects floating around, but it may not affect too much else.